packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/execute.ts

```diff
@@ -74,6 +74,40 @@ async function extractBinaryMessages(ctx: IExecuteFunctions) {
 		content: [...binaryMessages],
 	});
 }
+/**
+ * Fixes empty content messages in agent steps.
+ *
+ * This function is necessary when using RunnableSequence.from in LangChain.
+ * If a tool doesn't have any arguments, LangChain returns input: '' (empty string).
+ * This can throw an error for some providers (like Anthropic) which expect the input to always be an object.
+ * This function replaces empty string inputs with empty objects to prevent such errors.
+ *
+ * @param steps - The agent steps to fix
+ * @returns The fixed agent steps
+ */
+function fixEmptyContentMessage(steps: AgentFinish | AgentAction[]) {
+	if (!Array.isArray(steps)) return steps;
+
+	steps.forEach((step) => {
+		if ('messageLog' in step && step.messageLog !== undefined) {
+			if (Array.isArray(step.messageLog)) {
+				step.messageLog.forEach((message) => {
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+					if ('content' in message && Array.isArray(message.content)) {
+						// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+						(message.content as Array<{ input?: string | object }>).forEach((content) => {
+							if (content.input === '') {
+								content.input = {};
+							}
+						});
+					}
+				});
```

Comment:
Can we use types imported from Langchain here?
```suggestion
				step.messageLog.forEach((message: BaseMessage) => {
					if ('content' in message && Array.isArray(message.content)) {
						message.content.forEach((content: MessageContentComplex) => {
							if ('input' in content && content.input === '') {
								content.input = {};
							}
						});
					}
				});
```

Reply:
We could for `message: BaseMessage` but LC doesn't seem to export correct type for the message.content, so I still need to shim it like this